src/ucp/wireup/address.c

```diff
@@ -524,16 +566,22 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
             ucp_address_memchek(ptr, attr_len,
                                 &context->tl_rscs[dev->rsc_index].tl_rsc);
 
-            iface_addr_len = iface_attr->iface_addr_len;
+            if (flags & UCP_ADDRESS_PACK_FLAG_IFACE_ADDR) {
+                iface_addr_len = iface_attr->iface_addr_len;
+            } else {
+                iface_addr_len = 0;
+            }
             flags_ptr      = ucp_address_iface_flags_ptr(worker, ptr, attr_len);
             ptr           += attr_len;
             ucs_assert(iface_addr_len < UCP_ADDRESS_FLAG_EP_ADDR);
 
             /* Pack iface address */
             ptr    = ucp_address_pack_length(worker, ptr, iface_addr_len);
```

Comment:
can remove empty spaces before `=`  (before it was aligned with lower string, which is removed now)